### ISSUE TITLE
I can't "install" - Detected Connection error: [Errno -5] Name has no usable address

### ISSUE THREAD

**Issue by eph2** [alandtse/alexa_media_player#2604]
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/alandtse/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**

**Describe the bug**

I try to install the device on Home Assistant (on HACS is ok), I fill in my login details, double authentication key, it takes me to the page that generates the double authentication code, and after I say that everything went well, it takes me to a new window, where it returns the error below:

> Connection Error! Please try refreshing. If this persists, please report this error to [here](https://github.com/alandtse/alexa_media_player/issues):
> 
> [Errno -5] Name has no usable address

**To Reproduce**

1. Go to 'Settings'
2. Click on 'Devices & Services'
3. Click on 'Add devices'
4. Search for "Alexa Media Player"
5. Fill the inputs
6. Confirm the 2FA built-in code
7. Then the error occurs

**Expected behavior**

The device install and I can control my echo.

**Screenshots**

This is the screenshot (in Portuguese, sorry) to open the new window to authenticate
<img width="521" alt="Captura de Tela 2024-10-08 às 13 17 39" src="https://github.com/user-attachments/assets/4d665e54-87bb-4a60-aff2-62f409df32b9">
This button redirect to this URL:
http://homeassistant.local:8123/auth/alexamedia/proxy?config_flow_id=01J9PE6D6AF6JXEY488ZN4W8ZS&callback_url=http://homeassistant.local:8123/auth/alexamedia/callback?flow_id%3D01J9PE6D6AF6JXEY488ZN4W8ZS

**System details**

- Home Assistant version:2024.10.1
- alexa_media version (from `const.py` or HA startup log): 4.13.4
- alexapy version (from `pip show alexapy` in homeasssistant container or HA startup log): Can't find, but I don't remember installing it
- Is Amazon [2FA/2SV enabled](https://github.com/alandtse/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account)  (y/n): Y
- Amazon Domain: I tried with amazon.com, amazon.com.br and amazon.co.uk

**Debug Logs (alexa_media & alexapy)**
Este erro originou-se de uma integração personalizada.

Registrador: custom_components.alexa_media.config_flow
Fonte: custom_components/alexa_media/config_flow.py:1051
integração: alexa_media (documentação, problemas)
Ocorreu pela primeira vez: 13:07:46 (5 ocorrências)
Último registro: 13:16:33

Detected Connection error: [Errno -5] Name has no usable address

**Comment by danielbrunt57** [comment]
@eph2 After the Amazon/Alexa webpage opened, did you complete the email+password & OTP windows?  That should then return back to the HA page to fulfill the authentication requirements and the `Open website` window should then change to `Loading next step for Alexa Media Player`

**Comment by eph2** [comment]
Hey, @danielbrunt57. How are you? Thanks for your answer!

Sometimes this page (which asks for email, password and OTP) appears, sometimes it doesn't. When it does not show this "login screen", it automatically goes to the error page mentioned above.

When it appears, the following situations occur:

1. Email is automatically filled in, I click next, password is automatically filled in, I click next, it goes back to the email screen, and so on in an infinite loop.
2. Email is automatically filled in, I click next, password is automatically filled in, I click next and it shows the same error as in the original post about this issue.
3. I saw a "tip" here in the issues topics. So on the login screen, where the email is filled in, I click on the option to create an account. I fill in the data for the new account with my current data, it gives an error saying that the account already exists. I click on the option to log in, and the form shows the username and password on the same screen (already filled in). I click next, it asks for the OTP (which is already filled in), and when I click next it shows the same error as in the original post about this problem.

Another strange thing I noticed is that in HACS, after removing the plugin (to install it again), a screen appeared saying that this repository was no longer available. I tried to delete its address and add it again (manually) and the error occurred again.

After a while it became available. It seems to me that I'm having some DNS problem, route, something like that. I don't know if it's something with the plugin.

Also out of curiosity, I use ESPHome to program some devices, and I'm also having problems downloading from some repositories. I ended up seeing this after I opened this topic.

Does that make sense?
